src/foam/u2/wizard/StepWizardController.js

```diff
@@ -281,29 +281,48 @@ foam.CLASS({
         this.visitedWizardlets.indexOf(wizardlet) == -1 ? p
           : p.then(() => wizardlet.save()), Promise.resolve());
     },
-    function next() {
+    async function next() {
       // Save current wizardlet, and any save-able (isAvailable) but invisible
       // wizardlets that may exist in between this one and the next.
       // If it exists, load the next wizardlet
       // TODO: Just load next wizardlet instead of loading all in the beginning
+
+      // this.nextAvailable(wizardPosition, this.positionAfter.bind(this));
+      var currentWizardlet = this.wizardlets[this.wizardPosition.wizardletIndex];
+
+      // we want to save Facades since they could directly influence the availabities of other wizardlets
+      // if it is currently the "last" wizardlet in the flow
+      var isCurrentAlreadySaved = false;
+      if (
+        foam.u2.wizard.wizardlet.FacadeCapabilityWizardlet.isInstance(currentWizardlet)
+      ){
+        await currentWizardlet.save();
+        isCurrentAlreadySaved = true;
+      }
+
       var start = this.wizardPosition.wizardletIndex;
-      const nextScreen = this.nextAvailable(this.wizardPosition, this.positionAfter.bind(this));
+      let nextScreen = this.nextAvailable(this.wizardPosition, this.positionAfter.bind(this));
       var end = nextScreen ?
         nextScreen.wizardletIndex : this.wizardlets.length;
-      var p = Promise.resolve();
+
+      // if the current wizardlet is a facade wizardlet, then we need to save it first
+      // then we can recalculate end
```

Comment:
is this comment below where it should be?

Reply:
Removed.